worker/task/notification.go

```diff
@@ -0,0 +1,79 @@
+package task
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/frain-dev/convoy/util"
+
+	"github.com/frain-dev/convoy/config"
+	"github.com/frain-dev/convoy/datastore"
+	"github.com/frain-dev/convoy/notification"
+	"github.com/frain-dev/convoy/notification/email"
+	"github.com/frain-dev/convoy/notification/slack"
+	log "github.com/sirupsen/logrus"
+)
+
+func sendNotification(
+	ctx context.Context,
+	appRepo datastore.ApplicationRepository,
+	eventDelivery *datastore.EventDelivery,
+	g *datastore.Group,
+	smtpCfg *config.SMTPConfiguration,
+	status datastore.EndpointStatus,
+	failure bool,
+) {
+	app, err := appRepo.FindApplicationByID(ctx, eventDelivery.AppMetadata.UID)
+	if err != nil {
+		log.WithError(err).Error("failed to fetch application")
+		return
```

Comment:
@danvixent This should be `return err` and we shouldn't be logging here at all only return err.

Reply:
The ProcessEventDelivery function's flow cannot be disrupted by the error here, that is why I think it's best to log here.

What do you think?